File: dev-docs/snippets/cache/fetch-http.swift
import Foundation
import JsBaoClient

// Fetch from an HTTP endpoint with automatic caching keyed on the request.
func fetchHttp(client: JsBaoClient) async throws {
  // #region example
  let posts: [[String: Any]]? = try await client.cache.fetchHttp(
    method: "GET",
    path: "/api/posts",
    query: ["page": 1, "sort": "recent"],
    options: FetchCachedOptions(serverTimeoutMs: 5_000)
  )
  // #endregion example
  _ = posts
}

File: dev-docs/snippets/cache/fetch-http.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fetch from an HTTP endpoint with automatic caching keyed on the request.
export async function fetchHttp(client: JsBaoClient) {
  // #region example
  const posts = await client.cache.fetchHttp(
    {
      method: "GET",
      path: "/api/posts",
      query: { page: 1, sort: "recent" },
    },
    { serverTimeoutMs: 5_000 },
  );
  // #endregion example
  return posts;
}
